src/components/WorkNumberSelect.tsx

```diff
@@ -52,60 +59,95 @@ export default function WorkNumberSelect({
    */
   useEffect(() => {
     async function fetchActiveWorkNumbers() {
-      const response = await stanCore.FindWorkNumbers({
+      const response = await stanCore.FindWorkInfo({
         status: WorkStatus.Active,
       });
-      setWorks(response.works);
+      setWorks(
+        response.works.map((workInfo) => {
+          return {
+            workNumber: workInfo.workNumber,
+            workRequester: workInfo.workRequester
+              ? workInfo.workRequester.username
+              : "",
+            project: workInfo.project.name,
+          };
+        })
+      );
     }
     fetchActiveWorkNumbers();
   }, [setWorks]);
 
   useEffect(() => {
-    if (!workNumber) setSelectedWorkNumber("");
+    if (!workNumber) setSelectedWork(undefined);
     const work = works.find((work) => work.workNumber === workNumber);
     if (work) {
-      setSelectedWorkNumber(workNumber);
+      setSelectedWork(work);
     }
-  }, [workNumber, works]);
+  }, [workNumber, works, setSelectedWork]);
 
   /**
    * Callback for when the select changes
    */
   const handleWorkNumberChange = useCallback(
     (e: React.ChangeEvent<HTMLSelectElement>) => {
-      setSelectedWorkNumber(e.currentTarget.value);
+      setSelectedWork(
+        works.find((work) => work.workNumber === e.currentTarget.value)
+      );
       onWorkNumberChange?.(e.target.value);
     },
-    [onWorkNumberChange]
+    [onWorkNumberChange, setSelectedWork, works]
   );
 
   const validateWorkNumber = () => {
-    if (selectedWorkNumber === "") {
+    if (!selectedWork) {
       setError("SGP number is required");
     } else {
       setError("");
     }
   };
 
   return name ? (
-    <FormikSelect
-      label={label ?? ""}
-      name={name}
-      emptyOption={true}
-      onChange={handleWorkNumberChange}
-    >
-      {optionValues(works, "workNumber", "workNumber")}
-    </FormikSelect>
-  ) : (
-    <>
-      <Select
-        value={selectedWorkNumber}
-        onChange={handleWorkNumberChange}
+    <div className={"flex flex-col"}>
+      <FormikSelect
+        label={label ?? ""}
+        name={name}
         emptyOption={true}
-        onBlur={validateWorkNumber}
```

Comment:
Is there a reason this was removed? This will flash up the 'SGP number is required' error message onBlur. Think its still needed for components that arent using yup / formik

Reply:
Editing error, it seems. I have put that back 👍